test/integration/esm.spec.js

```diff
@@ -81,4 +84,22 @@ describe('esm', function() {
 
     expect(result, 'to have passed test count', 1);
   });
+
+  it('should throw an ERR_MODULE_NOT_FOUND and not ERR_REQUIRE_ESM if file imports a non-existing module', async function() {
+    const fixture =
+      'esm/type-module/test-that-imports-non-existing-module.fixture.js';
+
+    const [, promise] = invokeMochaAsync(
+      [resolveFixturePath(fixture), '--unhandled-rejections=warn'],
+      {stdio: 'pipe'}
+    );
```

Comment:
The test style you have chosen above (`should show file location when there is a syntax error in the test`) is much nicer than this one. I don't get any unhandled rejections neither.
IMO the tests are easier to read if you chose the same style for similar error test cases.
Can you change that, please?

Reply:
Good idea! Fixed.